### PR TITLE
added apache tika server

### PR DESCRIPTION
**Pull request by raghuramlavan**
I have added the tika server to the flake

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -105,6 +105,44 @@
             vendorSha256 = "sha256-f/DIAw8XWb1osfXAJ/ZKsB0sOmFnJincAQlfVHqElBE=";
           };
 
+          # Apache Tika server 
+          tika-server = with final; stdenv.mkDerivation rec {
+            name = "tika-server-1.26";
+  
+            src = fetchurl {
+              url = https://archive.apache.org/dist/tika/tika-server-1.26.jar;
```
Comment (happysalada) [inline flake.nix:113]: I would have prefered to have the src as an input. That makes easier updates.

File: flake.nix [hunk 2]
```diff
@@ -173,15 +212,24 @@
...
 
             options.services.ipfs-search = {
-              enable = mkOption {
-                type = types.bool;
-                default = false;
-                description = ''
-                  Whether to enable the ipfs-search service. It uses Rabbitmq, elastic-search, ipfs
-                '';
-              };
+                  enable = mkOption {
```
Comment (happysalada) [inline flake.nix:215]: formatting is a little weird, but don't worry, I'll fix that.

**Comment by happysalada** [comment]
Thank you!

File: flake.nix (hunk 1 above)
Comment (happysalada) [reply flake.nix:113]: I just realised that this was the jar file, so this won't work as an input. I'll leave the fetchurl as is then.

Comment (happysalada) [reply flake.nix:113]: If you are interested, I just found the source of the git repo https://github.com/apache/tika.git
I think ideally it should be built from source. I understand though that building java might be a pain, so no worries if you don't have time.

File: flake.nix [hunk 3]
```diff
@@ -105,6 +105,44 @@
...
+              gnupg
+            ];
+            nativeBuildInputs = [
+              makeWrapper
+            ];
+            installPhase = ''
+            echo "Installing.. "
+              mkdir -pv $out/share/java $out/bin
+              ls -l ${src}
+            cp ${src} $out/share/java/tika-server-1.27.jar
```
Comment (happysalada) [inline flake.nix:131]: the version here is 1.27 and in the url it's 1.26

File: flake.nix [hunk 4]
```diff
@@ -105,6 +105,44 @@
...
+            makeWrapper ${jre}/bin/java $out/bin/tika-server \
+              --add-flags "-jar $out/share/java/tika-server-1.27.jar" \
+              --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=on' \
+              --set _JAVA_AWT_WM_NONREPARENTING 1
+              '';
+            meta = {
+              homepage = "https://tika.apache.org/";
```
Comment (happysalada) [inline flake.nix:138]: I'm not sure that is what they are running. it says `tika-extractor` and it runs on port 8081
https://github.com/ipfs-search/ipfs-search/blob/master/docker-compose.yml
I couldn't see what is inside their docker image. tika server runs on port  9998 by default. I don't think it's the same thing.

Comment (raghuramlavan) [reply flake.nix:138]: ok. i might have made a mistake. will check please

Comment (raghuramlavan) [reply flake.nix:138]: there is a go module for extracting info from tika-server called tika-extractor. its a component. let me get back please.

Comment (happysalada) [reply flake.nix:138]: cool, let me know.
This is the one last piece before the ipfs-crawler can be run.

**Pull request merged**